Sources/ElasticSwift/Requests/DeleteRequest.swift

```diff
@@ -49,113 +53,57 @@ public class DeleteRequestBuilder: RequestBuilder {
         return self
     }
     
-    public func set(completionHandler: @escaping (DeleteResponse?, Error?) -> Void) -> Self {
+    public func set(completionHandler: @escaping (DeleteResponse?, Error?) -> ()) -> Self {
         self.completionHandler = completionHandler
         return self
     }
     
-    public func build() -> Request {
+    public func make() throws -> DeleteRequest {
         return DeleteRequest(withBuilder: self)
     }
-    
+
 }
 
 public class DeleteRequest: Request {
+   
+    public typealias ResponseType = DeleteResponse
     
-    let client: ESClient
+    public var completionHandler: ((DeleteResponse?, Error?) -> ())?
+    
+    public let client: ESClient
     let index: String
-    let type: String
+    let type: String?
     let id: String
     var version: Int?
-    var completionHandler: ((DeleteResponse?, Error?) -> Void)
     
     init(withBuilder builder: DeleteRequestBuilder) {
         self.client = builder.client
-        self.index = builder.index!
```

Comment:
Really dangerous to force unwrap non mandatory builder properties.
Resolved modifying those properties as nonOptional in the Builder and force setting them in the constructors